### DISCUSSION TITLE
Get multiple nodes at once

### DISCUSSION THREAD

**Discussion by Statkern15** [FreeOpcUa/python-opcua#1377] [Q&A] [answered]
For University I am working on a Siemens Sinumerik 840d Opc UA server. I need set multiple values at once. I've already figured `client.get_values()` is the way to go. However for that i first need the individual nodes. 
Right now im getting the node with `client.get_node()`. Since there are about 20-30 nodes I need to get this is a lot of copy paste. Is there a better way of getting multiple notes at once.

thanks in advance.

**Comment by AndreasHeine** [comment, marked as answer]
```python
import opcua

uri = "opc.tcp://127.0.0.1:4840"
client = opcua.Client(uri)
nodeids = ["ns=0;i=2256", "ns=0;i=2267"]
node_list = []
for nodeid in nodeids:
    node_list.append(client.get_node(nodeid))
client.get_values(node_list)
```

**Comment by Statkern15** [comment]
Thanks for the quick repley. So there is no other way than looping over them.

**Reply by AndreasHeine** [reply]
FAIL ^^

**Reply by Statkern15** [reply]
Alright thanks. I thought i've missed something.

**Reply by AndreasHeine** [reply]
forget the last one sorry...

```python
from opcua import Client, ua

uri = "opc.tcp://192.168.10.1:4840"
client = Client(uri)
client.connect()
nodeids = [ua.NodeId.from_string("ns=0;i=2256"), ua.NodeId.from_string("ns=0;i=2267")]
# low-level
results = client.uaclient.get_attributes(nodeids, ua.AttributeIds.Value)
print(results)
client.disconnect()
```
